core/docsite/source/mappers.html.md

```diff
@@ -45,14 +45,36 @@ Typically though, custom mappers will be used in more complex cases, when the un
 ``` ruby
 require 'rom/transformer'
 
-class MyMapper < ROM::Transformer
-  relation :users, as: :my_mapper
+class UserMapper < ROM::Transformer
+  relation :users, as: :users_mapper
+
+  # Each function in the pipeline is called in order and the row is sent as an argument.
+  map do
+    resolve_model
+    create_instance
+    # Any other functions you need.
+  end
+
+  # Find the model class for the row based on the content of its role
+  # field, then add it to the row's data for next function to use it.
+  def resolve_model(row)
+    model_for = ->(row) do
+      Object.const_get(Dry::Inflector.new.camelize(row[:role]))
+    end
 
-  map_array do
-    # define custom transformations here
+    row.to_h.merge(model: model_for.(row))
+  end
```

Comment:
I just realized this could be simplified like this:

```ruby
def resolve_model(row)
  [Inflector.classify(row[:role]), row]
end

def create_instance(model, row)
  model.new(row)
end
```

Notice `Inflector` which is a constant that we have under `ROM::Inflector`